src/transformers/utils/hub.py

```diff
@@ -200,11 +200,26 @@ def http_user_agent(user_agent: Union[Dict, str, None] = None) -> str:
     return ua
 
 
-def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None):
+def extract_commit_hash(resolved_file, commit_hash):
+    """
+    Extracts the commit hash from a resolved filename toward a cache file.
+    """
+    if resolved_file is None or commit_hash is not None:
+        return commit_hash
+
+    search = re.search(r"snapshots/([^/]+)/", resolved_file)
```

Comment:
potentially robustify this by looking for a hex string of length commit_hash_length

Reply:
(BTW could also potentially live in `huggingface_hub`)

Reply:
Reused REGEX_COMMIT_HASH from huggingface_hub here.